src/transform/src/nonnullable.rs

```diff
@@ -117,14 +116,3 @@ fn scalar_nonnullable(expr: &mut ScalarExpr, metadata: &RelationType) {
         }
     })
 }
-
-/// Transformations to aggregation functions, based on nonnullability of columns.
-fn aggregate_nonnullable(expr: &mut AggregateExpr, metadata: &RelationType) {
-    // An aggregate that is a count of non-nullable data can be replaced by a countall.
-    if let (AggregateFunc::Count, ScalarExpr::Column(c)) = (&expr.func, &expr.expr) {
-        if !metadata.column_types[*c].nullable && !expr.distinct {
-            expr.func = AggregateFunc::CountAll;
-            expr.expr = ScalarExpr::literal_null(ColumnType::new(ScalarType::Bool, true));
-        }
-    }
-}
```

Comment:
I might be confused, but I think we might still want this optimization. At least, something like it would be able to turn
```
select count(field)
```
into
```
select count(true)
```
which could then simplify through e.g. demand analysis.